api/datasets.go

```diff
@@ -436,6 +529,33 @@ func (h *DatasetHandlers) pullHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *DatasetHandlers) saveHandler(w http.ResponseWriter, r *http.Request) {
+	params := lib.SaveParams{}
+	err := parseProxyParams(r, &params)
+	if err != nil && err.Error() == "bad request" {
+		util.WriteErrResponse(w, http.StatusBadRequest, err)
+		return
+	}
+	if err == nil {
+		res, err := h.Save(r.Context(), &params)
```

Comment:
When we pass the full pre-computed parameters, we want to just consume them as is on the receiver end.